src/graph/validator/FetchVerticesValidator.cpp

```diff
@@ -16,238 +16,137 @@ namespace graph {
-      auto tagId = tagStatus.value();
+Expression *FetchVerticesValidator::rewriteIDVertex2Vid(const Expression *expr) {
+  auto *pool = qctx_->objPool();
+  auto matcher = [](const Expression *e) -> bool {
+    std::string lowerStr = e->toString();
+    folly::toLowerAscii(lowerStr);
+    return e->kind() == Expression::Kind::kFunctionCall && lowerStr == "id(vertex)";
+  };
+  auto rewriter = [pool](const Expression *e) -> Expression * {
+    UNUSED(e);
+    return InputPropertyExpression::make(pool, nebula::kVid);
+  };
+
+  return RewriteVisitor::transform(expr, std::move(matcher), std::move(rewriter));
+}
 
-      tags_.emplace(*tag, tagId);
-      auto schema = qctx_->schemaMng()->getTagSchema(space_.id, tagId);
-      if (schema == nullptr) {
-        LOG(ERROR) << "No schema found for " << *tag;
-        return Status::SemanticError("No schema found for `%s'", tag->c_str());
-      }
-      tagsSchema_.emplace(tagId, schema);
-    }
-  } else {
-    onStar_ = true;
-    const auto allTagsResult = qctx_->schemaMng()->getAllLatestVerTagSchema(space_.id);
-    NG_RETURN_IF_ERROR(allTagsResult);
-    const auto allTags = std::move(allTagsResult).value();
-    for (const auto &tag : allTags) {
+Status FetchVerticesValidator::validateTag(const NameLabelList *nameLabels) {
+  if (nameLabels == nullptr) {
+    // all tag
+    const auto &tagStatus = qctx_->schemaMng()->getAllLatestVerTagSchema(space_.id);
+    NG_RETURN_IF_ERROR(tagStatus);
+    for (const auto &tag : tagStatus.value()) {
       tagsSchema_.emplace(tag.first, tag.second);
     }
-    for (const auto &tagSchema : tagsSchema_) {
-      auto tagNameResult = qctx_->schemaMng()->toTagName(space_.id, tagSchema.first);
-      NG_RETURN_IF_ERROR(tagNameResult);
-      tags_.emplace(std::move(tagNameResult).value(), tagSchema.first);
+  } else {
+    auto labels = nameLabels->labels();
+    auto *schemaMng = qctx_->schemaMng();
+    for (const auto &label : labels) {
+      auto tagStatus = schemaMng->toTagID(space_.id, *label);
+      NG_RETURN_IF_ERROR(tagStatus);
+      auto tagID = tagStatus.value();
+      auto tagSchema = schemaMng->getTagSchema(space_.id, tagID);
+      if (tagSchema == nullptr) {
+        return Status::SemanticError("No schema found for `%s'", label->c_str());
+      }
+      tagsSchema_.emplace(tagID, tagSchema);
     }
   }
   return Status::OK();
 }
 
-Status FetchVerticesValidator::prepareVertices() {
-  auto *sentence = static_cast<FetchVerticesSentence *>(sentence_);
-  // from ref, eval when execute
-  if (sentence->vertices()->isRef()) {
-    srcRef_ = sentence->vertices()->ref();
-    auto result = checkRef(srcRef_, vidType_);
-    NG_RETURN_IF_ERROR(result);
-    inputVar_ = std::move(result).value();
-    return Status::OK();
+Status FetchVerticesValidator::validateYield(YieldClause *yield) {
+  auto pool = qctx_->objPool();
+  bool noYield = false;
+  if (yield == nullptr) {
+    // version 3.0: return Status::SemanticError("No YIELD Clause");
```

Comment:
Usually, we will write some todos for comment like this:
```
// TODO: Just compatible with previous version, this will be deprecated in version 3.0.
```

Reply:
done